compiler/linkname.go

```diff
@@ -75,6 +75,80 @@ func (n SymName) IsMethod() (recv string, method string, ok bool) {
 	return
 }
 
+// readLinknameFromComment reads the given comment to determine if it's a go:linkname
+// directive then returns the linkname information, otherwise returns nil.
+func readLinknameFromComment(pkgPath string, comment *ast.Comment) (*GoLinkname, error) {
+	if !strings.HasPrefix(comment.Text, `//go:linkname `) {
+		return nil, nil // Not a linkname compiler directive.
+	}
+
+	fields := strings.Fields(comment.Text)
+
+	// Check that the directive comment has both parts and is on the line by itself.
+	if len(fields) != 3 {
+		if len(fields) == 2 {
+			// Ignore one-argument form //go:linkname localname
+			// This is typically used with "insert"-style links to
+			// suppresses the usual error for a function that lacks a body.
+			// The "insert"-style links aren't supported by GopherJS so
+			// these bodiless functions have to be overridden in the natives anyway.
+			return nil, nil
+		}
+		return nil, fmt.Errorf(`gopherjs: usage requires 2 arguments: //go:linkname localname importpath.extname`)
+	}
```

Comment:
IMO, a switch would be more readable. The if-within-an-if makes me do a double take to be sure I understand what's going on.  But use your judgement.

Reply:
That's a good idea 👍